constraint/pkg/client/client.go

```diff
@@ -379,6 +387,17 @@ func (c *Client) RemoveTemplate(ctx context.Context, templ *templates.Constraint
 		return resp, err
 	}
 
+	for _, cstr := range c.constraints[artifacts.gk] {
+		if r, err := c.removeConstraintNoLock(ctx, cstr); err != nil {
```

Comment:
Should we remove individual entries upon success in case we exit early?

Reply:
Remove individual entries? Like from the cache? If so, removeConstraintNoLock does that (it's basically delete constraint but without the lock). Cleaning up the map is just for memory leaks.

Reply:
Ok you mean [here](https://github.com/open-policy-agent/frameworks/blob/a0f9a6b1e4e7f1fa76999e1b0ac9d78fdbeea75f/constraint/pkg/client/client.go#L568)? Missed that earlier, thanks!

Reply:
Yep!